core/testing.go

```diff
@@ -20,38 +23,44 @@ func MockConfig(t *testing.T) *Config {
 }
 
 // StartMockNode starts a mock Core node background process and returns it.
-func StartMockNode() *node.Node {
-	app := kvstore.NewApplication()
-	app.RetainBlocks = 10
+func StartMockNode(app types.Application) *node.Node {
 	return rpctest.StartTendermint(app, rpctest.SuppressStdout, rpctest.RecreateConfig)
 }
 
 func EphemeralMockEmbeddedClient(t *testing.T) Client {
-	nd := StartMockNode()
+	nd := StartMockNode(CreateKvStore(defaultRetainBlocks))
 	t.Cleanup(func() {
 		nd.Stop() //nolint:errcheck
 		rpctest.StopTendermint(nd)
 	})
 	return NewEmbeddedFromNode(nd)
 }
 
+// CreateKvStore creates a simple kv store app and gives the user
+// ability to set desired amount of blocks to be retained.
+func CreateKvStore(retainBlocks int64) *kvstore.Application {
```

Comment:
I don't really see a reason to pass in the `retainBlocks` param here when everything we do is using a package-globally accessible constant here but it's ok.

Reply:
You are right. The only motivation behind this func is to reduce the redundant code